src/app/download/download.service.ts

```diff
@@ -18,16 +16,46 @@ export class DownloadService {
     const file = this.adAuthoringState.getValue().file;
     const assetPath = file.name;
 
-    //TODO: If anything is undefined e.g the asset, make sure to send an error message
+    // TODO:
+    // If anything is undefined e.g the asset, make sure to send an error message
```

Comment:
We could disable the download button if required fields are not set (e.g. landingPageUrl and an asset must be set). Then we can have simple assertions here that the fields are defined. Feel free to add in a follow up PR.

Reply:
Completed this in the downlaodButtonDisable branch.